### PR TITLE
Course Activity Addition

### PR DESCRIPTION
**Pull request by Albee24**
Added the ability to filter students by not having course activity in the last seven days

### CODE REVIEW

File: StudentOutreach/Models/MessageFilter.swift [hunk 1]
```diff
@@ -9,7 +9,7 @@ import Foundation
 
 enum MessageFilter: CaseIterable, Identifiable {
     case notSubmitted, notGraded, scoredMoreThan, scoredLessThan, markedIncomplete, reassigned
-    case courseScoreLessThan, courseScoreMoreThan, courseScoreBetween, courseScoreEmpty, all
+    case courseScoreLessThan, courseScoreMoreThan, courseScoreBetween, courseScoreEmpty, NoCourseActivitySevenDays, all
```
Comment (ephraimkunz) [inline StudentOutreach/Models/MessageFilter.swift]: For consistency, let's start this with a lowercase letter: `noCourseActivitySevenDays`

File: StudentOutreach/Models/ViewModel.swift [hunk 2]
```diff
@@ -168,6 +168,7 @@ struct StudentAssignmentInfo: Hashable {
         let comps = formatter.personNameComponents(from: name)
         return comps?.givenName ?? name
     }
+    let lastCourseActivityAt: Date?
```
Comment (ephraimkunz) [inline StudentOutreach/Models/ViewModel.swift]: Let's move this up so it's above `firstName`. I generally like to keep computed properties separate from static properties.

**Review by ephraimkunz — COMMENTED** [review]
Thanks for the PR! This looks good to me generally, just a few comments about minor issues.

File: StudentOutreach/Models/MessageFilter.swift (hunk 1 above)
Comment (Albee24) [reply StudentOutreach/Models/MessageFilter.swift]: Made the change

File: StudentOutreach/Models/ViewModel.swift (hunk 2 above)
Comment (Albee24) [reply StudentOutreach/Models/ViewModel.swift]: Moved it up for consistency

**Review by ephraimkunz — APPROVED** [review]

**Pull request merged**